common/js/components/modal.js

```diff
@@ -1,37 +1,38 @@
-const modalToggleButtons = document.querySelectorAll(".modal-toggle-btn");
-
-modalToggleButtons.forEach((button) => {
-  button.addEventListener("click", () => {
+// 이벤트 위임 방식으로 모달 열기
+document.addEventListener("click", (e) => {
+  // 버튼 클릭 요소 또는 부모 요소 중에서 modal toggle btn 클래스 찾기
+  const button = e.target.closest(".modal-toggle-btn");
+  if (button) {
     const modalId = button.dataset.modalTarget;
     if (!modalId) return;
 
     const modal = document.getElementById(modalId);
-    if (!modal) return;
-
-    modal.style.display = "flex";
-  });
+    if (modal) {
+      modal.style.display = "flex";
+    }
+  }
 });
 
-document.querySelectorAll(".modal").forEach((modal) => {
-  // 모달 내의 x 클릭시 모달 닫기
-  modal
-    .querySelector(".modal-close")
-    .addEventListener("click", () => closeModal(modal));
-  // 모달 내의 취소 버튼 클릭시 모달 닫기
-  modal
-    .querySelector(".cancel-btn")
-    .addEventListener("click", () => closeModal(modal));
-  // 모달 배경 클릭시 모달 닫기
-  modal.addEventListener("click", function (e) {
-    e.target === modal && closeModal(modal);
-  });
+// 이벤트 위임방식 모달 닫기
+document.addEventListener("click", (e) => {
+  // X 버튼 클릭 클릭 시 닫기
+  if (e.target.closest(".modal-close")) {
+    const modal = e.target.closest(".modal");
+    if (modal) closeModal(modal);
+  }
+
+  // 취소 버튼 클릭 시 닫기
+  if (e.target.closest(".cancel-btn")) {
+    const modal = e.target.closest(".modal");
+    if (modal) closeModal(modal);
+  }
```

Comment:
이 부분은 같은 로직을 반복하니, 하나의 조건문으로 해결 가능할 것 같아요!
```suggestion
document.addEventListener("click", (e) => {
  // X 버튼 클릭 클릭 시 닫기
  if (e.target.closest(".modal-close, .cancle-btn")) {
    const modal = e.target.closest(".modal");
    if (modal) closeModal(modal);
  }
```